src/encoded/static/components/auditmatrixv1.js

```diff
@@ -0,0 +1,740 @@
+    context: PropTypes.object.isRequired,
+    /** Callback to retrieve row categories from matrix data */
+    rowCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to retrieve subcategories from matrix data */
+    rowSubCategoryGetter: PropTypes.func.isRequired,
+    /** Callback to map row category query values */
+    mapRowCategoryQueries: PropTypes.func.isRequired,
+    /** Callback to map subcategory query values */
+    mapSubCategoryQueries: PropTypes.func.isRequired,
+};
+
+
+/**
+ * Determines the sorting order of the audits as they appear in the matrix.
+ */
+const auditOrderKey = [
+    'no_audits',
+    'audit.WARNING.category',
+    'audit.NOT_COMPLIANT.category',
+    'audit.ERROR.category',
+    'audit.INTERNAL_ACTION.category',
+];
+
+/**
+ * Determines the sorting order of the "No audits" subcategories.
+ */
+const noAuditOrderKey = [
+    'no errors, compliant, and no warnings',
+    'no errors and compliant',
+    'no errors',
+    'no audits',
+];
+
+/**
+ * Audit matrix rowCategory colors.
+ */
+const auditColors = ['#009802', '#e0e000', '#ff8000', '#cc0700', '#a0a0a0'];
+
+/**
+ * Maps audit keys to human-readable names.
+ */
+const auditNames = {
+    no_audits: 'No audits',
+    'audit.WARNING.category': 'Warning',
+    'audit.NOT_COMPLIANT.category': 'Not Compliant',
+    'audit.ERROR.category': 'Error',
+    'audit.INTERNAL_ACTION.category': 'Internal Action',
+};
+
+
+/**
+ * Map query values to a query-string component actually used in audit matrix row category link
+ * queries.
+ * @param {string} rowCategory Row category value to map
+ * @param {object} rowCategoryBucket Matrix search result row bucket object
+ *
+ * @return {string} mapped row category query
+ */
+const mapRowCategoryQueriesAudit = (rowCategory, rowCategoryBucket) => {
+    if (rowCategoryBucket.key === 'no_audits') {
+        return 'audit.ERROR.category%21=%2A&audit.NOT_COMPLIANT.category%21=%2A&audit.WARNING.category%21=%2A&audit.INTERNAL_ACTION.category%21=%2A';
+    }
+    return `${rowCategoryBucket.key}=%2A`;
+};
+
+
+/**
+ * Map query values to a query-string component actually used in audit matrix subcategory link
+ * queries.
+ * @param {string} subCategory Subcategory value to map
+ * @param {string} subCategoryQuery Subcategory query value to map
+ * @param {object} rowCategoryBucket Matrix search result row bucket object
+ *
+ * @return {string} mapped subcategory query
+ */
+const mapSubCategoryQueriesAudit = (subCategory, subCategoryQuery, rowCategoryBucket) => {
+    let query;
+    if (rowCategoryBucket.key === 'no_audits') {
+        switch (subCategoryQuery) {
+        case 'no audits':
+            query = 'audit.WARNING.category%21=%2A&audit.NOT_COMPLIANT.category%21=%2A&audit.ERROR.category%21=%2A&audit.INTERNAL_ACTION.category%21=%2A';
+            break;
+        case 'no errors':
+            query = 'audit.ERROR.category%21=%2A';
+            break;
+        case 'no errors, compliant, and no warnings':
```

Comment:
Do you need to consider the case like: 'no errors, no warnings and compliant '? Like, will ordering always be like this?

Reply:
That would be up to wranglers. They came up with these categories.